searchlib/src/vespa/searchlib/attribute/imported_search_context.cpp

```diff
@@ -170,8 +185,8 @@ void ImportedSearchContext::fetchPostings(bool strict) {
     assert(!_fetchPostingsDone);
     _fetchPostingsDone = true;
     _target_search_context->fetchPostings(strict);
-    if (strict) {
-        makeMergedPostings();
+    if (strict || _target_attribute.getConfig().fastSearch()) {
```

Comment:
Making a large synthetic posting lists can be more expensive if the non-strict iterator is only going to get a few seek() calls.  We don't have the necessary information here since we don't have a proper cost model.

Reply:
That is correct, but is was the most pragmatic approximation that I could think of this morning. I think that at least for gemini it should hold quite well.